examples/with-viem/src/contracts.ts

```diff
@@ -0,0 +1,104 @@
+import * as path from "path";
+import * as dotenv from "dotenv";
+
+import { createAccount } from "@turnkey/viem";
+import { TurnkeyClient } from "@turnkey/http";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import {
+  createWalletClient,
+  createPublicClient,
+  http,
+  type Account,
+} from "viem";
+import { goerli } from "viem/chains";
+import { print } from "./util";
+import abi from "./abi.json";
+
+// Load environment variables from `.env.local`
+dotenv.config({ path: path.resolve(process.cwd(), ".env.local") });
+
+async function main() {
+  const turnkeyClient = new TurnkeyClient(
+    {
+      baseUrl: process.env.BASE_URL!,
+    },
+    new ApiKeyStamper({
+      apiPublicKey: process.env.API_PUBLIC_KEY!,
+      apiPrivateKey: process.env.API_PRIVATE_KEY!,
+    })
+  );
+
+  const turnkeyAccount = await createAccount({
+    client: turnkeyClient,
+    organizationId: process.env.ORGANIZATION_ID!,
+    privateKeyId: process.env.PRIVATE_KEY_ID!,
+  });
+
+  const client = createWalletClient({
+    account: turnkeyAccount as Account,
+    chain: goerli,
+    transport: http(
+      `https://goerli.infura.io/v3/${process.env.INFURA_API_KEY!}`
+    ),
+  });
+
+  const address = client.account.address;
+  print("Address:", address);
+
+  const account = client?.account?.address;
+  const hash = await client.writeContract({
+    abi,
+    address: "0x85111DF47D97Cc1aB3e76889fCF08594d053E85d",
+    functionName: "mint",
+    chain: goerli,
+    account,
+  });
+
+  print("Mint transaction hash:", hash);
+
+  const publicClient = createPublicClient({
+    transport: http("https://rpc.ankr.com/eth_goerli"),
+    chain: goerli,
+  });
+
+  const { request } = await publicClient.simulateContract({
+    abi,
+    address: "0x85111DF47D97Cc1aB3e76889fCF08594d053E85d",
+    functionName: "mint",
+    chain: goerli,
+    account,
```

Comment:
Similarly, if `account` is included, it'll lead to the same error as above ^. If omitted, simulation actually runs, but with the assumption that the destination address is null/zero (since this is using the `publicClient`):

```
/Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/utils/errors/getContractError.js:28
    return new contract_js_1.ContractFunctionExecutionError(cause, {
           ^

ContractFunctionExecutionError: The contract function "mint" reverted.

Error: MintToZeroAddress()
 
Contract Call:
  address:   0x85111DF47D97Cc1aB3e76889fCF08594d053E85d
  function:  mint()

Docs: https://viem.sh/docs/contract/simulateContract.html
Version: viem@1.10.0
    at getContractError (/Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/utils/errors/getContractError.js:28:12)
    at simulateContract (/Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/actions/public/simulateContract.js:44:41)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at main (/Users/andrew/tkhq/code/sdk/examples/with-viem/src/contracts.ts:64:23) {
  details: undefined,
  docsPath: '/docs/contract/simulateContract',
  metaMessages: [
    'Error: MintToZeroAddress()',
    ' ',
    'Contract Call:',
    '  address:   0x85111DF47D97Cc1aB3e76889fCF08594d053E85d\n' +
      '  function:  mint()'
  ],
  shortMessage: 'The contract function "mint" reverted.',
  version: 'viem@1.10.0',
  cause: /Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/utils/errors/getContractError.js:21
          cause = new contract_js_1.ContractFunctionRevertedError({
                  ^
  
  ContractFunctionRevertedError: The contract function "mint" reverted.
  
  Error: MintToZeroAddress()
  
  
  Version: viem@1.10.0
      at getContractError (/Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/utils/errors/getContractError.js:21:17)
      at simulateContract (/Users/andrew/tkhq/code/sdk/node_modules/.pnpm/viem@1.10.0_typescript@5.1.3/node_modules/viem/dist/cjs/actions/public/simulateContract.js:44:41)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at main (/Users/andrew/tkhq/code/sdk/examples/with-viem/src/contracts.ts:64:23) {
    details: undefined,
    docsPath: undefined,
    metaMessages: [ 'Error: MintToZeroAddress()', '' ],
    shortMessage: 'The contract function "mint" reverted.',
    version: 'viem@1.10.0',
    data: {
      abiItem: [Object],
      args: undefined,
      errorName: 'MintToZeroAddress'
    },
    reason: undefined,
    signature: undefined
  },
  ...
```